pkg/blockstorage/azure/azuredisk.go

```diff
@@ -376,95 +357,98 @@ func (s *AdStorage) SnapshotGet(ctx context.Context, id string) (*blockstorage.S
 	if err != nil {
 		return nil, errors.Wrapf(err, "SnapshotsClient.Get: Failure in parsing snapshot ID %s", id)
 	}
-	snap, err := s.azCli.SnapshotsClient.Get(ctx, rg, name)
+	snapRes, err := s.azCli.SnapshotsClient.Get(ctx, rg, name, nil)
 	if err != nil {
 		return nil, errors.Wrapf(err, "SnapshotsClient.Get: Failed to get snapshot with ID %s", id)
 	}
 
-	return s.snapshotParse(ctx, snap), nil
+	return s.snapshotParse(ctx, snapRes.Snapshot), nil
 }
 
 func (s *AdStorage) VolumeParse(ctx context.Context, volume interface{}) (*blockstorage.Volume, error) {
-	vol, ok := volume.(azcompute.Disk)
+	vol, ok := volume.(armcompute.Disk)
 	if !ok {
-		return nil, errors.New(fmt.Sprintf("Volume is not of type *azcompute.Disk, volume: %v", volume))
+		return nil, errors.New(fmt.Sprintf("Volume is not of type *armcompute.Disk, volume: %v", volume))
 	}
 	encrypted := false
-	if vol.DiskProperties.EncryptionSettingsCollection != nil &&
-		vol.DiskProperties.EncryptionSettingsCollection.Enabled != nil {
-		encrypted = *vol.DiskProperties.EncryptionSettingsCollection.Enabled
+	if vol.Properties.EncryptionSettingsCollection != nil &&
+		vol.Properties.EncryptionSettingsCollection.Enabled != nil {
+		encrypted = *vol.Properties.EncryptionSettingsCollection.Enabled
 	}
 	tags := map[string]string{"": ""}
 	if vol.Tags != nil {
-		tags = azto.StringMap(vol.Tags)
+		tags = blockstorage.StringMap(vol.Tags)
 	}
-	az := azto.String(vol.Location)
-	if z := azto.StringSlice(vol.Zones); len(z) > 0 {
-		az = az + "-" + z[0]
+	az := blockstorage.StringFromPtr(vol.Location)
+	if z := vol.Zones; len(z) > 0 {
+		az = az + "-" + *(z[0])
 	}
 
 	return &blockstorage.Volume{
 		Type:         s.Type(),
-		ID:           azto.String(vol.ID),
+		ID:           blockstorage.StringFromPtr(vol.ID),
 		Encrypted:    encrypted,
-		SizeInBytes:  azto.Int64(vol.DiskSizeBytes),
+		SizeInBytes:  blockstorage.Int64(vol.Properties.DiskSizeBytes),
 		Az:           az,
 		Tags:         blockstorage.MapToKeyValue(tags),
-		VolumeType:   string(vol.Sku.Name),
-		CreationTime: blockstorage.TimeStamp(vol.DiskProperties.TimeCreated.ToTime()),
-		Attributes:   map[string]string{"Users": azto.String(vol.ManagedBy)},
+		VolumeType:   string(*vol.SKU.Name),
```

Comment:
[nil pointer] we should guard for `nil` pointer here before dereference. Perhaps we can take `encrypted` as an example:
```go
	if vol.Properties.EncryptionSettingsCollection != nil &&
		vol.Properties.EncryptionSettingsCollection.Enabled != nil {
		encrypted = *vol.Properties.EncryptionSettingsCollection.Enabled
	}
```